public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -30,16 +32,17 @@ export const RestoreActivitiesPanel: React.FC<RestoreActivitiesPanelProps> = ({
   const [flyout, setFlyout] = useState(false);
 
   useEffect(() => {
-    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
-
-    let getStatusInterval: ReturnType<typeof setInterval>;
+    context?.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
 
-    if (stage.slice(0, 4) !== "Done") {
-      getStatusInterval = setInterval(() => {
+    if (stage !== "Done (100%)" || indices.length < restoreCount) {
```

Comment:
This process needs to start and stop automatically, which led me to use "useEffect".  Since useEffect runs again when state changes theres no way around it.  Thus the array to store them in outside of useEffect was the way I eventually found to ensure that auto refresh would absolutely stop every time 100% is achieved.  I can experiment with it for sure but due to time constraints can we tag this for a refactor when time allows?  It works just as it should atm.

Reply:
That may not be a perfect react explanation^